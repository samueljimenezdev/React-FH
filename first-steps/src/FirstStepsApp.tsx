import { ItemCounter } from "./shopping-cart/ItemCounter";

export interface ItemInCart {
    productName: string,
    quantity: number
}

const itemsInCart: ItemInCart[] = [
    { productName: 'Nintendo', quantity: 0 },
    { productName: 'Xbox', quantity: 0 },
    { productName: 'Play Station', quantity: 0 }
]

export function FirstStepsApp() {
    return (
        <>
            <h1>Carrito de compra</h1>
            {
                itemsInCart.map(({ productName, quantity }) => (
                    <ItemCounter key={productName} name={productName} quantity={quantity} />
                ))
            }
        </>

    )
}