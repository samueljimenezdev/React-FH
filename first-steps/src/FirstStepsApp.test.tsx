import { render, screen } from "@testing-library/react";
import { beforeEach, describe, expect, test, vi } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";


const mockItemCounter = vi.fn((props: unknown) => {
    return (
        <div data-testid="ItemCounter" />
    )
})

vi.mock('./shopping-cart/ItemCounter', () => ({
    ItemCounter: (props: unknown) => mockItemCounter(props)
}))

// vi.mock('./shopping-cart/ItemCounter', () => ({
//     ItemCounter: () =>
//         <div
//             data-testid="ItemCounter"
//         />,
// }))

describe('FirstStepApss', () => {

    beforeEach(() => {
        vi.clearAllMocks()
    })

    test('should match snapshot ', () => {
        const { container } = render(<FirstStepsApp />)
        expect(container).toMatchSnapshot()
    })

    test('should render the correct number of ItemCounter elements', () => {
        render(<FirstStepsApp />)
        console.log(screen.debug())
        const itemCounters = screen.getAllByTestId('ItemCounter');
        expect(itemCounters.length).toBe(3)
    })

    test('should render ItemCounter with correcto props', () => {
        render(<FirstStepsApp />)
        expect(mockItemCounter).toHaveBeenCalledTimes(3);
        expect(mockItemCounter).toHaveBeenCalledWith({ name: 'Nintendo', quantity: 0 });
        expect(mockItemCounter).toHaveBeenCalledWith({ name: 'Xbox', quantity: 0 });
        expect(mockItemCounter).toHaveBeenCalledWith({ name: 'Play Station', quantity: 0 })

    })
})