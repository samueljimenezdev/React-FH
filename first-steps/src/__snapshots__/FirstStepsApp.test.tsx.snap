// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepApss > should match snapshot  1`] = `
<div>
  <h1>
    Carrito de compra
  </h1>
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
</div>
`;
